### PR TITLE
Prevent Toil from zipping entire dirs

### PR DESCRIPTION
**Pull request by jeffrey856**
Added a check to not allow user to execute toil in a dir such as /tmp or /var.

### CODE REVIEW

File: src/toil/resource.py [hunk 1]
```diff
@@ -287,11 +287,21 @@ def _load(cls, path):
         else:
             # This is a simple user script (with possibly a few helper files)
             rootDir = path
-        with ZipFile(file=bytesIO, mode='w') as zipFile:
-            for dirName, _, fileList in os.walk(path):
-                for fileName in fileList:
-                    fullPath = os.path.join(dirName, fileName)
-                    zipFile.write(fullPath, os.path.relpath(fullPath, rootDir))
+        blacklistDir = ['/tmp', '/var', '/etc', '/bin', '/sbin', '/home', '/dev', '/sys', '/usr', '/run']
```
Comment (mr-c) [inline src/toil/resource.py]: let's call this the skipdirList, that'd be clearer

Comment (jeffrey856) [reply src/toil/resource.py]: got it! updated it in new commit

File: src/toil/resource.py [hunk 2]
```diff
@@ -287,11 +287,21 @@ def _load(cls, path):
...
+        if path not in skipdirList:
+            with ZipFile(file=bytesIO, mode='w') as zipFile:
+                for dirName, _, fileList in os.walk(path):
+                    for fileName in fileList:
+                        try:
+                            fullPath = os.path.join(dirName, fileName)
+                            zipFile.write(fullPath, os.path.relpath(fullPath, rootDir))
+                        except IOError:
+                            log.critical('Cannot access and read the file at address: %s' % fullPath)
+                            sys.exit()
```
Comment (adamnovak) [inline src/toil/resource.py]: This should be `sys.exit(1)`. As is, we will exit with success.

File: src/toil/resource.py [hunk 3]
```diff
@@ -287,11 +287,21 @@ def _load(cls, path):
...
+        else:
+            log.critical("Couldn't package the directory at %s for hot deployment. Would recommend to create a \
+                subdirectory (ie %s/MYDIR_HERE/)" % (path, path))
+            sys.exit()
```
Comment (adamnovak) [inline src/toil/resource.py]: This should also be `sys.exit(1)`.

File: src/toil/resource.py [hunk 4]
```diff
@@ -287,11 +287,21 @@ def _load(cls, path):
...
+        skipdirList = ['/tmp', '/var', '/etc', '/bin', '/sbin', '/home', '/dev', '/sys', '/usr', '/run']
+        if path not in skipdirList:
+            with ZipFile(file=bytesIO, mode='w') as zipFile:
+                for dirName, _, fileList in os.walk(path):
+                    for fileName in fileList:
+                        try:
+                            fullPath = os.path.join(dirName, fileName)
+                            zipFile.write(fullPath, os.path.relpath(fullPath, rootDir))
+                        except IOError:
+                            log.critical('Cannot access and read the file at address: %s' % fullPath)
```
Comment (adamnovak) [inline src/toil/resource.py]: Say `path` instead of `address` here.

**Pull request merged**